### PR TITLE
fix: output null for NaN/Infinity float/double values in JSON

### PR DESCRIPTION
**Pull request by rophy**
## Summary

- OLR previously output bare `nan`/`inf`/`-inf` tokens for IEEE 754 special values in BINARY_FLOAT/BINARY_DOUBLE columns, producing invalid JSON
- Default behavior now outputs `null` (matches JavaScript's `JSON.stringify(NaN)` convention and Debezium+LogMiner parity)
- Adds `json-number-type` format option: `0` = null (default), `1` = quoted strings (`"NaN"`, `"Infinity"`, `"-Infinity"`) for future Debezium versions with NaN support ([DBZ-806](https://issues.redhat.com/browse/DBZ-806))
- Adds IEEE 754 edge case test rows (NaN, ±Infinity, negative zero, subnormal, min/max normal)

## Known issues

- Subnormal values (test ID 12) are decoded as half the correct value due to a separate bug in `decodeFloat`/`decodeDouble` — tracked in #5

## Test plan

- [x] Debezium twin-test: 14 PASS, 0 FAIL (number-precision excluded due to pre-existing precision issues #3, #5)
- [x] NaN/Inf rows (IDs 8-10) match between LogMiner and OLR paths
- [ ] Redo log regression tests (CI)

## Release Notes

* **New Features**
  * Added `json-number-type` configuration option to control how special floating-point numbers (NaN, Infinity) are serialized in JSON output. Choose between null (default) or string representation.

* **Documentation**
  * Updated documentation with usage guidance and compatibility notes for the new option.

* **Tests**
  * Expanded test coverage for special numeric values and edge cases.

### CODE REVIEW

File: src/OpenLogReplicator.cpp [hunk 1]
```diff
@@ -766,6 +768,13 @@ namespace OpenLogReplicator {
                 redoThreadFormat = static_cast<Format::REDO_THREAD_FORMAT>(val);
             }
 
+            if (formatJson.HasMember("json-number-type")) {
+                const uint val = Ctx::getJsonFieldU(configFileName, formatJson, "json-number-type");
+                if (val > 1)
+                    throw ConfigurationException(30001, "bad JSON, invalid \"json-number-type\" value: " + std::to_string(val) + ", expected: one of {0, 1}");
+                jsonNumberType = static_cast<Format::JSON_NUMBER_TYPE>(val);
+            }
```
[the remark this answers is not in the document]
Comment (rophy) [reply src/OpenLogReplicator.cpp]: Valid observation, but this would be inconsistent with how existing format options work. Options like `rid`, `xid`, `message`, `scn`, etc. are all parsed regardless of `format.type` and silently ignored by builders that don't use them. Adding a format-type guard only for `json-number-type` would be a one-off special case. If we want strict validation, it should be done consistently across all format options — but that's a separate concern.
